jax/_src/test_util.py

```diff
@@ -468,6 +468,47 @@ def is_device_tpu(version: int | None = None, variant: str = "") -> bool:
     return "v6 lite" in device_kind
   return expected_version in device_kind
 
+def device_kind_match(device_patterns: str | Sequence[str]):
+  if isinstance(device_patterns, list):
+    device_patterns = tuple(device_patterns)
+  elif not isinstance(device_patterns, tuple):
+    device_patterns = (device_patterns,)  # type: ignore
+
+  current_device = jax.devices()[0].device_kind
+  for device_pattern in device_patterns:
+    if device_pattern in current_device:
+      return device_pattern
+  return None
+
+def skip_if_device_kind_matches(device_patterns: str | Sequence[str]):
+  """A decorator for test methods to skip the test when run on specified devices."""
+  matching_pattern = device_kind_match(device_patterns)
+
+  reason = f"Device kind matches \"{matching_pattern}\""
+  def skip(test_method):
+    return unittest.skipIf(matching_pattern is not None, reason)(test_method)
+  return skip
+
+def skip_if_exceeds_shared_memory_on(device_patterns: str | Sequence[str]):
+  """Skip if a kernel exceeds shared memory on a specified device."""
+  def skip(test_method):
+    @functools.wraps(test_method)
+    def test_method_wrapper(self, *args, **kwargs):
+      try:
+        return test_method(self, *args, **kwargs)
+      except Exception as e:
+        mosaic_gpu_pattern = "kernel exceeds available shared memory"
+        triton_pattern = "Shared memory size limit exceeded"
```

Comment:
This is very mosaic GPU/triton specific. Maybe it would be better to have a `skip_if_errors(device_patterns, messages)` and specialize the message in the pallas tests?